### DISCUSSION TITLE
ODE stepper

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#21870] [Q&A General] [answered]
I am using the BDF2 and AB2PredictorCorrector set for time-stepping. I also use postprocessors to monitor values of variables and energies etc. I noted something weird in the Postprocessors. Here is what I have in relevnt blocks:
```
[Executioner]
  type = Transient
  solve_type = ''PJFNK'
  scheme = 'BDF2'
 [./TimeStepper]
   type = AB2PredictorCorrector
  e_max = 5.e-3
  e_tol = 5.e-3
 ....
[../]
[]
```
But the postprocessors return 0 for any execute_on other than 'nonlinear', eg
```
[Postprocessors]
  [./<mx>]
  type = ElementAverageValue
  variable = mag_y
  evaluate_on = 'nonlinear'
  [../]
[]
```
Any other value for evaluate_on gives me zero. This seems to be related to the BDF2/AB2PredictorCorrector. Am I doing something wrong here or missing something?

Thanks,
Olle

**Comment by GiudGiud** [comment, marked as answer]
Hello

Should be execute_on not evaluate_on. 
Is mag_y a nonlinear variable or an auxvariable?

Time stepping and time integration schemes are independent, you should be able to swap them out almost freely. Could you please see if BDF2 or AB2PC is the problem?

Guillaume

**Reply by heinono1** [reply]
Thanks, Guillaume. I did have the correct 'execute_on' in the input file - I incorrectly typed 'evaluate_on' in the posting.

Swapping crank-nicolson or nothing (default implicit euler) for BDF2 gives errors (unknown time integrator) even though I follow the syntax in https://mooseframework.inl.gov/source/timesteppers/AB2PredictorCorrector.html. So not sure what is going on.

Cheers,
Olle

**Reply by heinono1** [reply]
Oh, mag_y is a nonlinear variable.

**Reply by GiudGiud** [reply]
You should be able to remove the AB2 and use BDF2 on its own.

Tbh I think AB2 is going to be the problem here
